engine/gameobject/proto/gameobject/gameobject_ddf.proto

```diff
@@ -61,6 +61,12 @@ message ComponentPropertyDesc
     optional dmPropertiesDDF.PropertyDeclarations property_decls = 3;
 }
 
+message ComponenTypeDesc
+{
+    required uint64 name_hash = 1;
+    required uint32 max_count = 2;
+}
+
```

Comment:
I use name_hash instead of string name. I think this is what we prefer over string all the time if it's possible.